crates/oxc_transformer/src/jsx/refresh.rs

```diff
@@ -279,10 +262,19 @@ impl<'a, 'ctx> Traverse<'a> for ReactRefresh<'a, 'ctx> {
                         false,
                     ),
                 );
-                self.extra_statements
-                    .entry(id_binding.symbol_id)
-                    .or_insert(ctx.ast.vec())
-                    .push(statement);
+
+                let mut target_ancestor = ctx.ancestor(1);
+                for ancestor in ctx.ancestors().skip(2) {
+                    if !matches!(
+                        ancestor,
+                        Ancestor::VariableDeclarationDeclarations(_)
+                            | Ancestor::ExportNamedDeclarationDeclaration(_)
+                    ) {
+                        break;
+                    }
+                    target_ancestor = ancestor;
+                }
+                self.ctx.statement_injector.insert_after(&target_ancestor, statement);
```

Comment:
I think this line is not correct:

https://github.com/oxc-project/oxc/blob/c0574494289bd7efe6bf1b4708751c7db1889d89/crates/oxc_transformer/src/jsx/refresh.rs#L250

Wouldn't code like this cause a panic?

```js
const {length} = () => {};
```

Reply:
We have no changes to your mentioned code, But you are right, I will fix it in another PR

Reply:
I raised this here because I thought the two were connected. But I was wrong. Parent of a `VariableDeclarator` is always a `VariableDeclaration`, regardless of the type of the `VariableDeclarator`. So my point is not relevant here.

But I still think this block of code can be simplified. Something like:

```rs
// Get `Statement` containing this `VariableDeclarator`.
// Its parent is always a `VariableDeclaration`.
// That's the `Statement`, unless grandparent is an `ExportNamedDeclaration`.
let target_ancestor = match ctx.ancestor(2) {
    ancestor @ Ancestor::ExportNamedDeclarationDeclaration(_) => ancestor,
    _ => ctx.ancestor(1).
};
```